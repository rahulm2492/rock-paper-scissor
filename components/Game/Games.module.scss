.game {
	border: 0;
	min-width: 320px;
	h1 {
		background-color: #152a43;
		font-size: 20px;
		margin: 0 0 10px;
		padding: 20px;
		text-align: center;
	}
}
.challenge > div {
	margin: 0 auto;
}
.footer {
	background-color: #4B6A90;
	margin-top: 20px;
	padding: 10px;
}
.modes {
	margin: 20px 0;
}

