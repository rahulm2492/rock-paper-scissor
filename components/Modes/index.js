import React, { PropTypes } from 'react';
import styles from './Modes.module.scss';

const Modes = ({ currentMode, onToggleMode }) => {
	return(<div className={styles.Modes}>
		<span className="label">{currentMode?.label}</span><br />
		<button
			onClick={onToggleMode}
		>
			CHANGE MODE
		</button>
	</div>
)};


export default Modes;