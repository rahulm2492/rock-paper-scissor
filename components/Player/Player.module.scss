  
.Player {
	width: 100%;
	text-align: center;
	
}
.image {
	width: 50%;
	margin: 0 auto;
}

.score {
	border: 1px solid #fff;
	border-radius: 3px;
	display: inline-block;
	min-width: 60px;
	padding: 4px;
}
.label {
	font-weight: bold;
}