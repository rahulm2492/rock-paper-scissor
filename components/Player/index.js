import React from 'react';
import styles from './Player.module.scss';

import Weapon from '../Weapons';

const Player = ({ label, weapon, loading, score }) => (
	<div className={styles.Player}>
		<div>
			<span className="label">{label}</span>
		</div>
		{weapon && <Weapon
			weapon={weapon}
			loading={loading}
		/>}
		<div>
			<span className="score">{score} PT{score > 1 && 'S'}</span>
		</div>
	</div>
);

export default Player;