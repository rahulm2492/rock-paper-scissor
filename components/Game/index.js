import React, { useState, useEffect } from "react";
import styles from "./Games.module.scss";
import Modes from "../Modes";
import { getComputerWeapon, getResult } from "../../utils/gameLogic";
import Player from "../Player";
import WeaponList from "../WeaponsList";


const Game = (props) => {
  const { weapons, modes } = props.data;
  const getComputerMove = getComputerWeapon(weapons.length);
  const getGameResult = getResult(weapons);

  const [currentMode, toggleMode] = useState(0);
  const [gameRound, setGameCounter] = useState(0);

  const [player1Weapon, setPlayer1Weapon] = useState(undefined);
  const [player2Weapon, setPlayer2Weapon] = useState(undefined);
  const [score1, setScore1] = useState(0);
  const [score2, setScore2] = useState(0);

  const { player1Label, player2Label } = modes[currentMode];

  useEffect(() => {
    const result = getGameResult(player1Weapon, player2Weapon);
    result == 1 && setScore1(score1 + 1);
	result == 0 && setScore2(score2 + 1);
	setGameCounter(gameRound+1);
  }, [player1Weapon, player2Weapon]);

  useEffect(() => {
    setScore1(0);
    setScore2(0);
    setPlayer1Weapon(undefined);
	setPlayer2Weapon(undefined);
	setGameCounter(0);
  }, [currentMode]);

  const weaponSelected = (yourWeapon) => {
    setPlayer1Weapon(yourWeapon);
    const computerWeapon = getComputerMove();
    setPlayer2Weapon(computerWeapon);
  };

  return (
    <div className={styles.game}>
      <h1>ROCK, PAPER, SCISSORS</h1>
  <h5>Round: {gameRound}</h5>
      <div className={styles.modes}>
        <Modes
          onToggleMode={() => toggleMode(!currentMode * 1)}
          currentMode={modes[currentMode]}
        />
      </div>

      <div className={styles.challenge}>
        <Player
          label={player1Label}
          weapon={weapons[player1Weapon]}
          score={score1}
        />
        {!!currentMode && (
          <button
            title="Play"
            onClick={() => weaponSelected(getComputerMove())}
          >
            Play
          </button>
        )}
        <Player
          label={player2Label}
          weapon={weapons[player2Weapon]}
          score={score2}
        />
      </div>

      <div className="footer">
        <WeaponList
          weapons={weapons}
		  onSelectWeapon={(weapon) => weaponSelected(weapon)}
		  mode={currentMode}
        />
      </div>
    </div>
  );
};

export default Game;
