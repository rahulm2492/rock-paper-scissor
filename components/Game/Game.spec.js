import { shallow, mount } from "enzyme";

import Game from "./index";
import Modes from "../Modes";
import Player from "../Player";

const data = {
  weapons: [
    {
      beats: ["Scissors"],
      icon: "rock.svg",
      label: "Rock",
    },
    {
      beats: ["Rock"],
      icon: "paper.svg",
      label: "Paper",
    },
    {
      beats: ["Paper"],
      icon: "scissors.svg",
      label: "Scissors",
    },
  ],
  modes: [
    {
      label: "PLAYER VS COMPUTER",
      player1Label: "PLAYER",
      player2Label: "COMPUTER",
    },
    {
      label: "COMPUTER VS COMPUTER",
      player1Label: "COMPUTER 1",
      player2Label: "COMPUTER 2",
    },
  ],
};
/** @test {Game Component} */
describe("Game Component", () => {
  it("should render without crashing", () => {
    const wrapper = shallow(<Game label="test" />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render Mode of Game", () => {
    const wrapper = mount(<Game data={data} />);
    const inst = wrapper.instance();
    const mode = wrapper.find(Modes);
    expect(mode.find('.label').text()).toEqual(data.modes[0].label);
    mode.find('button').simulate('click');
    expect(mode.find('.label').text()).toEqual(data.modes[1].label);
    
});
it("should render Players of Game", () => {
    const wrapper = mount(<Game data={data} />);
    const inst = wrapper.instance();
    const player1 = wrapper.find(Player).at(0);
    expect(player1.find('span').at(0).text()).toEqual(data.modes[0].player1Label);
    const player2 = wrapper.find(Player).at(1);
    expect(player2.find('span').at(0).text()).toEqual(data.modes[0].player2Label);

    
});
});
