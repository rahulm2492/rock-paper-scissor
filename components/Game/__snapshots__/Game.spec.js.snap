// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Component should render without crashing 1`] = `ShallowWrapper {}`;
